packages/block-editor/src/components/provider/test/use-block-sync.js

```diff
@@ -14,7 +14,17 @@ import { render } from '@testing-library/react';
 import useBlockSync from '../use-block-sync';
 import withRegistryProvider from '../with-registry-provider';
 import * as blockEditorActions from '../../../store/actions';
+
 import { store as blockEditorStore } from '../../../store';
+jest.mock( '../../../store/actions', () => {
```

Comment:
Why this change was needed?

Reply:
Because of a nuance of importing – we used to spy on the actions module entry by replacing the `module[actionName]` via `jest.spyOn`. That was fine, but now there's an additional destructuring happening in `store.js`:

```js
import * as allActions from './actions';
const { __experimentalUpdateSettings, ...actions } = allActions;
```

The `spyOn` call was executed after the destructuring and had no effect on the registered store. `jest.mock()` call is, I believe, hoisted and occurs before the destructuring above – which means it has an effect on the store.